node.go

```diff
@@ -87,12 +97,69 @@ func (node *emptyNode) isEmpty() bool {
 
 var _ ProofNode = (*node)(nil)
 
+func (node *node) GetHash() []byte {
+	parent, ok := node.parent.(*interiorNode)
+	if !ok {
+		return nil
+	}
+	switch parent.leftChild.(type) {
+	case *emptyNode:
+		if &parent.leftChild.(*emptyNode).node == node {
+			return parent.leftHash
```

Comment:
The ```ProofNode``` should only expose the hashes of the pruned subtrees. This seems to be doing the opposite. I believe this should return ```parent.rightHash``` if the node on the path is the ```leftChild```. The same goes for the ```userLeafNode``` case.